tests/func/test_data_status.py

```diff
@@ -161,7 +141,221 @@ def test_subdir(M, tmp_dir, scm):
+        "committed": {"added": M.unordered("foobar", join("dir", ""))},
+        "not_in_cache": M.unordered("foobar", join("dir", "")),
+        "git": M.dict(),
+    }
+
+    assert dvc.data_status(granular=True, untracked_files="all") == {
+        **EMPTY_STATUS,
+        "untracked": M.unordered("foobar.dvc", "dir.dvc", ".gitignore"),
+        # "committed": {
+        #   "added": M.unordered(
+        #       "foobar", join("dir", "foo"), join("dir", "bar")
+        # )},
+        "committed": {"added": M.unordered("foobar", join("dir", ""))},
+        "uncommitted": {
+            "unknown": M.unordered(join("dir", "foo"), join("dir", "bar"))
+        },
+        # even though we don't have cache obj, we can repurpose
+        # what we have in the workspace
+        "not_in_cache": M.unordered(
+            "foobar",
+            join("dir", ""),
+        ),
+        "git": M.dict(),
+    }
+
+
+def test_missing_cache_missing_workspace(M, tmp_dir, dvc, scm):
+    tmp_dir.dvc_gen({"dir": {"foo": "foo", "bar": "bar"}})
+    tmp_dir.dvc_gen("foobar", "foobar")
+    for path in [dvc.odb.repo.path, "dir", "foobar"]:
+        remove(path)
+
+    assert dvc.data_status(untracked_files="all") == {
+        **EMPTY_STATUS,
+        "untracked": M.unordered("foobar.dvc", "dir.dvc", ".gitignore"),
+        "uncommitted": {"deleted": M.unordered("foobar", join("dir", ""))},
+        "committed": {"added": M.unordered("foobar", join("dir", ""))},
+        "not_in_cache": M.unordered("foobar", join("dir", "")),
+        "git": M.dict(),
+    }
+
+    assert dvc.data_status(granular=True, untracked_files="all") == {
+        **EMPTY_STATUS,
+        "untracked": M.unordered("foobar.dvc", "dir.dvc", ".gitignore"),
+        "uncommitted": {"deleted": M.unordered("foobar", join("dir", ""))},
+        "committed": {"added": M.unordered("foobar", join("dir", ""))},
+        "not_in_cache": M.unordered("foobar", join("dir", "")),
+        "git": M.dict(),
+    }
+
+
+def test_git_committed_missing_cache_workspace_exists(M, tmp_dir, dvc, scm):
+    tmp_dir.dvc_gen({"dir": {"foo": "foo", "bar": "bar"}}, commit="add dir")
+    tmp_dir.dvc_gen("foobar", "foobar", commit="add foobar")
+    remove(dvc.odb.local.path)
+
+    assert dvc.data_status(untracked_files="all") == {
+        **EMPTY_STATUS,
+        "not_in_cache": M.unordered("foobar", join("dir", "")),
+        "git": M.dict(),
+        # TODO: clarify what is unchanged
+        "unchanged": M.unordered("foobar", join("dir", "")),
+    }
+    assert dvc.data_status(granular=True) == {
+        **EMPTY_STATUS,
+        "not_in_cache": M.unordered(
+            "foobar",
+            join("dir", ""),
+        ),
+        "uncommitted": {
+            "unknown": M.unordered(join("dir", "foo"), join("dir", "bar"))
+        },
+        "git": M.dict(),
+        # TODO: clarify what is unchanged
+        "unchanged": M.unordered("foobar", join("dir", "")),
```

Comment:
`unchanged` may be broken in some `Not in cache` edgecases, due to `unknown`. This may require to be reconsidered, but otherwise should be fine.